src/index.js

```diff
@@ -8,26 +8,31 @@ const options = Object.freeze({
 
 const NightModeToggle = ({ size, checked, onChange, ...extraProps }) => {
   const ref = useRef();
-  const [progress, setProgress] = useState(() => 0);
+  const [progress, setProgress] = useReducer((_,next) => next, 0);
+
   useEffect(() => {
     if (progress >= 0.5) {
-      if (checked) {
+      if (ref.current && checked) {
         ref.current.anim.pause();
-      } else if (ref.current.anim.isPaused) {
+      } else if (ref.current && ref.current.anim.isPaused) {
         ref.current.anim.play();
       }
-    } else if (!checked) {
+    } else if (ref.current && !checked) {
       ref.current.anim.pause();
     }
   }, [checked, progress]);
+
   useEffect(() => (!!checked && ref.current.anim.play()) || undefined, []);
-  const [eventListeners] = useState(() => [
-    {
+
+  const eventListeners = useMemo(
```

Comment:
Is there any benefit to using `useMemo` as opposed to `useState` in this instance? Genuinely curious!